benchmarks/perf-tool/okpt/io/dataset.py

```diff
@@ -64,9 +65,9 @@ class HDF5DataSet(DataSet):
     <https://github.com/erikbern/ann-benchmarks#data-sets>`_
     """
 
-    def __init__(self, dataset_path: str, context: Context):
+    def __init__(self, dataset_path: str, context: Context, custom_context=None):
```

Comment:
What is meant by custom_context?

Reply:
in case of custom context the type of the content is set by the user as opposite to few predefined ones like 'neighbors', 'train' and 'test'. As we filter datasets is not something standard my choice was in favor of more configurable approach.